### PR TITLE
Generalize OCI spec root

### PR DESCRIPTION
**Pull request by saschagrunert**
We now generalize and document the OCI `Spec` root structure. This means
that some fields have been added and other are now optional.

All corresponding usages of the new spec format have been changed and
tests have been adapted.

### CODE REVIEW

File: src/container/builder_impl.rs
Comment (Furisto) [inline src/container/builder_impl.rs:132]: Can you change this to "no process spec" here and everywhere else? My first reaction to "no process" in an error message would be to assume that the container process has exited, not that some part of the spec is missing.

Comment (saschagrunert) [reply src/container/builder_impl.rs]: Yep, changed as suggested.

**Review by Furisto — APPROVED** [review]

**Comment by Furisto** [comment]
@saschagrunert Looks good to me. As @utam0k has taken the lead on this, I will wait for his review before merge.

**Review by utam0k — APPROVED** [review]
great!

**Pull request merged**
